File: backend/src/main/java/ecrow/backend/business/abstracts/RequirementService.java
package ecrow.backend.business.abstracts;

import ecrow.backend.core.utilities.results.DataResult;
import ecrow.backend.core.utilities.results.Result;
import ecrow.backend.entities.concretes.Requirement;
import ecrow.backend.entities.dtos.RequirementAddDto;
import ecrow.backend.entities.dtos.RequirementSatisfiedUpdateDto;

import java.util.List;

public interface RequirementService {
    DataResult<List<Requirement>> getAll();
    DataResult<Requirement> getById(Integer id);
    DataResult<List<Requirement>> getByFkTransactionId(Integer itemTransactionId);
    Result deleteById(Integer id);

    Result add(RequirementAddDto requirementAddDto);
    Result updateSatisfied(RequirementSatisfiedUpdateDto requirementSatisfiedUpdateDto);
    Result updateSatisfiedTrue(Integer id);

}

File: frontend/src/services/RequirementService.js
import axios from "axios";

export default class RequirementService{
    getAll() {
        return axios.get("http://localhost:8080/api/paymentMethod/getAll")
    }

    getById(id) {
        return axios.get(`http://localhost:8080/api/paymentMethod/getById?id=${id}`)
    }

    getByTransactionId(transactionId){
        return axios.get(`http://localhost:8080/api/requirement/getByFkTransactionId?itemTransactionId=${transactionId}`)
    }

    deleteById(id) {
        return axios.delete(`http://localhost:8080/api/requirement/deleteById?id=${id}`)
    }

    add(values){
        return axios.post("http://localhost:8080/api/requirement/add", values)
    }

    updateSatisfied(values){
        return axios.patch("http://localhost:8080/api/requirement/updateSatisfied", values)
    }

    updateSatisfiedTrue(id){
        return axios.patch(`http://localhost:8080/api/requirement/updateSatisfiedTrue?id=${id}`)
    }
}